js/pro/kucoinfutures.js

```diff
@@ -0,0 +1,968 @@
+                //                     endpoint: "wss://push-private.kucoin.com/endpoint",
+                //                     protocol: "websocket",
+                //                     encrypt: true,
+                //                     pingTimeout: 10000
+                //                 }
+                //             ],
+                //             token: "2neAiuYvAU61ZDXANAGAsiL4-iAExhsBXZxftpOeh_55i3Ysy2q2LEsEWU64mdzUOPusi34M_wGoSf7iNyEWJ1UQy47YbpY4zVdzilNP-Bj3iXzrjjGlWtiYB9J6i9GjsxUuhPw3BlrzazF6ghq4Lzf7scStOz3KkxjwpsOBCH4=.WNQmhZQeUKIkh97KYgU0Lg=="
+                //         }
+                //     }
+                //
+            } else {
+                response = await this.futuresPublicPostBulletPublic ();
+            }
+            client.resolve (response, messageHash);
+            // const data = this.safeValue (response, 'data', {});
+            // const instanceServers = this.safeValue (data, 'instanceServers', []);
+            // const firstServer = this.safeValue (instanceServers, 0, {});
+            // const endpoint = this.safeString (firstServer, 'endpoint');
+            // const token = this.safeString (data, 'token');
+        }
+        return await future;
+    }
+
+    requestId () {
+        // TODO
+        const requestId = this.sum (this.safeInteger (this.options, 'requestId', 0), 1);
+        this.options['requestId'] = requestId;
+        return requestId;
+    }
+
+    async subscribe (negotiation, topic, messageHash, method, symbol, params = {}) {
+        await this.loadMarkets ();
+        // const market = this.market (symbol);
+        const data = this.safeValue (negotiation, 'data', {});
+        const instanceServers = this.safeValue (data, 'instanceServers', []);
+        const firstServer = this.safeValue (instanceServers, 0, {});
+        const endpoint = this.safeString (firstServer, 'endpoint');
+        const token = this.safeString (data, 'token');
+        const nonce = this.requestId ();
+        const query = {
+            'token': token,
+            'acceptUserMessage': 'true',
+            // 'connectId': nonce, // user-defined id is supported, received by handleSystemStatus
+        };
+        const url = endpoint + '?' + this.urlencode (query);
+        // const topic = '/market/snapshot'; // '/market/ticker';
+        // const messageHash = topic + ':' + market['id'];
+        const subscribe = {
+            'id': nonce,
+            'type': 'subscribe',
+            'topic': topic,
+            'response': true,
+        };
+        const subscription = {
+            'id': nonce.toString (),
+            'symbol': symbol,
+            'topic': topic,
+            'messageHash': messageHash,
+            'method': method,
+        };
+        const request = this.extend (subscribe, params);
+        const subscriptionHash = topic;
+        return await this.watch (url, messageHash, request, subscriptionHash, subscription);
```

Comment:
@pcriadoperez Hey I thought you might be able to help here, nothing is being returned from this method with `watchTicker`, when you turn on `--verbose` though, you get the response from the exchange, but for some reason the parsed response isn't being returned